### DISCUSSION TITLE
Do you need to run next build for every new blog post?

### DISCUSSION THREAD

**Discussion by Malthehave** [vercel/next.js#12641] [Help] [answered]
Hello! 

This has been on my mind for some time now.
I'm wondering if you have to build your site again for each blog post. 

Let's say I have a headless CMS where I publish some blogs posts from time to time. Do I then need to run `next build` everytime I publish a new post?

I took the tutorial from https://nextjs.org/learn/basics/. And the `getStaticProps` function seems to only run at build time. How would it then be possible to read the posts that I publish in the future?

I guess I'm missing something here. Would love to hear your feedback!

**Comment by Timer** [comment, marked as answer]
You may be looking for fallback behavior:
https://nextjs.org/docs/basic-features/data-fetching#the-fallback-key-required

**Reply by Malthehave** [reply]
That does sound like the thing I need yes.
In terms of the blog scenario, would using a fallback be the preferred method?

**Reply by Timer** [reply]
Yup! This is what we use on Vercel's blog and exactly what we built it for.

**Reply by Malthehave** [reply]
Sweet! Thanks for clearing that up for me :)

**Reply by iuritorres** [reply]
the link is broken :(

is this one you're talking about?
https://nextjs.org/docs/app/building-your-application/data-fetching/incremental-static-regeneration
